### DISCUSSION TITLE
Metadata issue ?

### DISCUSSION THREAD

**Discussion by BH-NL** [savonet/liquidsoap#2777] [User Support]
Hi,
I have now spent many hours to get the metadata working. I want Liquidsoap  send the icy metadata to icecast. Im running icecast port 8000 on windows server 2016 and liquidsoap 2.0.7 on that same server. The incoming icy metadata (headers) from source "s" on port 8003 is visible in the logs of Liquidsoap.  I use "metadata.map" to add the metadata to the source. This seems to work, when I print the metadata, it is added, but besides "title"it does not show up in icecast. When I use list.add(("title" the title is showing up in icecast, but when I try list.add(("genre" or any other piece of metadata it doesn't work. I have tried also with 2.1.3, same result. Is this a Liquidsoap/windows issue?

```
headers = ref ([])

def on_connect(new_headers) =
  headers := new_headers
end

s = input.harbor("stream",port=8003,password="hackme",icy=true,on_connect=on_connect)

source.on_metadata(s, fun (n) -> print("Got meta 1: #{n}")) 

def add_dj_metadata(metadata) =
  print("************************METADATA-1*************************")	
  headers = !headers
  if headers["ice-genre"] != "" then
	icegenre = headers["ice-genre"]
	# Update metadata
	print("metadata added")
	list.add(("genre", "#{icegenre}"), metadata)
  else
    metadata
  end
end

s = metadata.map(add_dj_metadata, s)

source.on_metadata(s, fun (o) -> print("Got meta 2: #{o}")) 
```

Regards

**Comment by toots** [comment]
Hi @BH-NL,

Sorry that this is getting frustrating.

Do you mean that when adding `"genre"` in the mapping function, it does not show up in the `""Got meta 2: ..."` log or that it does not show up on `icecast`?

Also wanted to add a note here that metadata are filtered before being exported in order to avoid leaking potentially sensitive ones. This shouldn't be an issue for your since `"genre"` is allowed but I wanted to add it for documentation purposes. The setting is:
```
settings.encoder.metadata.export.set([
  "artist", "title", "album", "genre", "date",
  "tracknumber", "comment", "track",
  "year", "dj", "next", "apic",
  "metadata_block_picture", "coverart"])
```

**Reply by BH-NL** [reply]
Hi Toots,

Thanks for your reply. It does not show up in icecast.
I added the print commands for my own use to see if the metadata was added correctly.
I also added a print on the input of the output.icecast, all added metadata is there.
If I add a static "genre = music, " to the output.icecast, icecast is showing the genre.

With the metadata.map I can overwrite "title" and "artist", that is sended to icecast correct, all other like "genre" and "dj" don't seem to work.

I added the setting.encoder.metadata.export to my script, just to try, but that didn't make a difference.

Unless I'm doing something wrong it looks like the output.icecast is besides "title" and "artist" not processing the metadata comming from the input.

Regards

**Reply by toots** [reply]
Hi. I think I'm missing some of the logic here, seems like there are a couple of missing negations in your explanation.. 🙂

One thing I just realized reading the operator's doc again (https://www.liquidsoap.info/doc-2.1.3/reference.html#output.icecast), `genre` is a static value that you set in the output. It don't think that it can be changed dynamically via metadata. This is a limitation coming from icecast, not us.

**Reply by BH-NL** [reply]
Hi, sorry if my explanation is not clear, my english is not that good.

I assumed that the purpose of adding dynamic metadata to the source was to send it to a program like icecast. So that was wrong.
I'm glad to know it can not work because of a limitation of icecast, now I can spend my time working on a other solution to send the metadata 🙂 .
Thanks for your help!

Regards
